cli/pcluster/configure/easyconfig.py

```diff
@@ -360,21 +368,15 @@ def prompt_os(self):
 
     def prompt_instance_types(self):
         """Ask for master_instance_type and compute_instance_type (if necessary)."""
-        ec2_client = boto3.client("ec2")
-        instance_type_offerings = [
-            offering["InstanceType"]
-            for offering in ec2_client.describe_instance_type_offerings()["InstanceTypeOfferings"]
-        ]
-
         self.master_instance_type = prompt(
             "Master instance type",
-            lambda x: x in get_supported_instance_types() and x in instance_type_offerings,
+            lambda x: x in get_supported_instance_types() and _is_instance_type_supported_for_head_node(x),
```

Comment:
I'd call the functions in the opposite order, because the `_is_instance_type_supported_for_head_node`is a static validation, the other one is a boto3 call, in this way we can fail fast.